python/tk_multi_loader/delegate_publish_list.py

```diff
@@ -0,0 +1,380 @@
+            self._format_folder(model_index, widget)
+            
+        else:
+            # this is a publish!
+            self._format_publish(model_index, widget)
+
+    def _format_folder(self, model_index, widget):
+        """
+        Formats the associated widget as a folder item.
+        
+        :param model_index: Model index to process
+        :param widget: widget to adjust
+        """        
+        sg_data = shotgun_model.get_sg_data(model_index)
+        field_data = shotgun_model.get_sanitized_data(model_index, 
+                                                      shotgun_model.ShotgunModel.SG_ASSOCIATED_FIELD_ROLE)
+
+        # examples of data. In the examples below, when we say "node", this 
+        # means a node in the tree structure in the left hand side tree
+        
+        # intermediate node which isn't a link 
+        # field_data: {'name': 'sg_asset_type', 'value': 'Character' }
+        # sg_data:    None
+        
+        # intermediate node which is an entity
+        # field_data: {'name': 'sg_sequence', 'value': {'type': 'Sequence', 'id': 11, 'name': 'bunny_080'}}
+        # sg_data:    None
+        
+        # leaf node:  
+        # field_data: {'name': 'code', 'value': 'aaa_0030'}
+        # sg_data: {'code': 'aaa_00030',
+        #           'description': 'Created by the Shotgun Flame exporter.',
+        #           'id': 1662,
+        #           'image': 'https://....',
+        #           'project': {'id': 289, 'name': 'Climp', 'type': 'Project'},
+        #           'sg_sequence': {'id': 202, 'name': 'aaa', 'type': 'Sequence'},
+        #           'sg_status_list': 'wtg',
+        #           'type': 'Shot'}
+        
+        field_value = field_data["value"]
+        
+        # by default, just display the value 
+        main_text = field_value
+        small_text = ""
+
+        if isinstance(field_value, dict) and "name" in field_value and "type" in field_value:
+            # intermediate node with entity link
+            main_text = "<b>%s</b> <b style='color:#2C93E2'>%s</b>" % (field_value["type"], field_value["name"])
+
+        elif isinstance(field_value, list):
+            # this is a list of some sort. Loop over all elements and extract a comma separated list.
+            # this can be a multi link field but also a field like a tags field or a non-entity link type field.
+            formatted_values = []
+            formatted_types = set()
+            
+            for v in field_value:
+                if isinstance(v, dict) and "name" in v and "type" in v:
+                    # This is a link field
+                    if v.get("name"):
+                        formatted_values.append(v.get("name"))
+                        formatted_types.add(v["type"])
+                else:
+                    formatted_values.append(str(v))
+            
+            if len(formatted_types) == 0:
```

Comment:
good point will fix! I think previously, there was something else going on in that if/else clause, which later on got removed, leaving us with this. Nice to clean up! :)